h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -1008,17 +1013,70 @@ AutoBuffer putUdp(UDP.udp type, int senderPort){
    *
    * @param type type of the UDP datagram
    */
-  AutoBuffer putUdp (UDP.udp type) {
-    return putUdp(type, H2O.H2O_PORT); // Outgoing port is always the sender's (me) port
+  AutoBuffer putUdp(UDP.udp type) {
+    return putUdp(type, H2O.H2O_PORT);
+  }
+
+  /**
+   * Select last 15 bytes from the jvm boot start time and return it as char
+   */
+  public static char createUniqueId(long jvmStartTime){
+    byte availableBits = 15;
+    int bitMask = (1 << availableBits) - 1;
+
+    // we set the lower 15 bits, the 1 bit is currently set to 0 which means we are not a client
+    return (char) (jvmStartTime & bitMask);
+  }
+
+
+  private static char CLIENT_CODE_MASK = (char) (1 << 15);
+  private static char CLIENT_DECODE_MASK = (char) (Math.pow(2, 15) - 1);
+  /**
+   * Calculate node metadata from Current's node information. We use start of jvm boot time and information whether
+   * we are client or not. We combine these 2 information and create a char(2 bytes) with this info in a single variable.
+   */
+  static char calculateNodeUniqueMeta() {
```

Comment:
Ideally would be calculated just once during startup after the arguments are parsed. This would give us a little bit more straightforward workflow.

Reply:
Implemented

Reply:
Good idea, done